src/net/unixsock_readmsg_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// go:build aix darwin dragonfly freebsd linux netbsd openbsd solaris
+
+package net
+
+import (
+	"os"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func TestUnixConnReadMsgUnixSCMRightsCloseOnExec(t *testing.T) {
+	if !testableNetwork("unix") {
+		t.Skip("not unix system")
+	}
+
+	scmFile, err := os.Open(os.DevNull)
+	if err != nil {
+		t.Fatalf("file open: %v", err)
+	}
+	defer scmFile.Close()
+
+	rights := syscall.UnixRights(int(scmFile.Fd()))
+	fds, err := syscall.Socketpair(syscall.AF_LOCAL, syscall.SOCK_STREAM, 0)
+	if err != nil {
+		t.Fatalf("Socketpair: %v", err)
+	}
```

Comment:
@tklauser I don't have to ability to leave comments in Gerrit, but you left the following comment:
> We need to defer close the sockets here, otherwise we might leak file descriptors:
  defer syscall.Close(fds[0])
  defer syscall.Close(fds[1])

This is actually NOT true, and it would be a major bug to add it. This is because `os.NewFile` just wraps the existing fd instead of duplicating it so you cannot do both `syscall.Close(fds[0])` and `writeFile.Close()`, as it would try to close the same fd twice. Likewise for `fds[1]` and `readFile`.

In short, @HowJMay please do NOT apply the aforementioned suggestion.

Reply:
We're using Gerrit for code reviews, so please leave your future comments regarding the CL there. See https://golang.org/doc/contribute for more information on how to create and account.

I've replied to the comment in question.